### ISSUE TITLE
Created USB device does not boot due to bootx64.efi file being 0-bytes sized

### ISSUE THREAD

**Issue by rathann** [WoeUSB/WoeUSB#138]
## Describe the bug ##

Bootable USB device created with WoeUSB does not boot.

## To Reproduce ##

1. `sudo woeusb --device pl_windows_7_home_premium_with_sp1_x64_dvd_u_676540.iso /dev/sdc`
2. Move the USB device to the target machine
3. Try booting
4. Device does not boot

## Expected behavior ##

Created USB device should boot (in UEFI mode, in particular).

## Teminal output ##

```your-woeusb-output-here:
WoeUSB v5.2.4
==============================
Info: Mounting source filesystem...
Info: Wiping all existing partition table and filesystem signatures in /dev/sdc...
/dev/sdc: 8 bytes were erased at offset 0x00000200 (gpt): 45 46 49 20 50 41 52 54
/dev/sdc: 8 bytes were erased at offset 0x1e77ffe00 (gpt): 45 46 49 20 50 41 52 54
/dev/sdc: 2 bytes were erased at offset 0x000001fe (PMBR): 55 aa
/dev/sdc: calling ioctl to re-read partition table: Success
Info: Ensure that /dev/sdc is really wiped...
Info: Creating new partition table on /dev/sdc...
Info: Creating target partition...
Info: Making system realize that partition table has changed...
Info: Wait 3 seconds for block device nodes to populate...
mkfs.fat 4.2 (2021-01-31)
mkfs.fat: Warning: lowercase labels might not work properly on some systems
Info: Mounting target filesystem...
Info: Copying files from source media...
Warning: Source media seems to be Windows 7-based with EFI support, applying workaround to make it support UEFI booting
Info: Installing GRUB bootloader for legacy PC booting support...
Installing for i386-pc platform.
Installation finished. No error reported.
Info: Installing custom GRUB config for legacy PC booting...
Info: Done :)
Info: The target device should be bootable now
Info: Unmounting and removing "/tmp/woeusb-source-20241007-124001-Monday.HIoSks"...
Info: Unmounting and removing "/tmp/woeusb-target-20241007-124001-Monday.7CpAG5"...
Info: You may now safely detach the target device
```

## Environment ##

- OS: Fedora 40
- Bash: 5.2.26
- WoeUSB: 5.2.4

## Additional context ##
The `sources/install.wim` file in the ISO seems to contain the `bootmgfw.efi` file at `1/Windows/Boot/EFI/bootmgfw.efi`. After extracting that one and replacing the 0-bytes sized `bootx64.efi` at `efi/boot`, the device becomes bootable.

In other words, the code at https://github.com/WoeUSB/WoeUSB/blob/master/sbin/woeusb#L1878 makes no effort to check that it actually extracted the bootloader.

**Comment by brlin-tw** [comment]
Thanks for the report, yes the extraction is pretty basic.  Why is the bootmgfw.efi file be in that path in that image?

**Comment by rathann** [comment]
> Thanks for the report, yes the extraction is pretty basic. Why is the bootmgfw.efi file be in that path in that image?

I have no idea. This is an unmodified image downloaded a few years ago from Microsoft.

**Comment by Ittihadyya** [comment]
bump. this issue still exists.
